test/framework/cluster_topology_helpers.go

```diff
@@ -65,13 +66,16 @@ type UpgradeClusterTopologyAndWaitForUpgradeInput struct {
 	EtcdImageTag                            string
 	DNSImageTag                             string
 	MachineDeployments                      []*clusterv1.MachineDeployment
+	MachinePools                            []*expv1.MachinePool
 	KubernetesUpgradeVersion                string
 	WaitForMachinesToBeUpgraded             []interface{}
+	WaitForMachinePoolToBeUpgraded          []interface{}
 	WaitForKubeProxyUpgrade                 []interface{}
 	WaitForDNSUpgrade                       []interface{}
 	WaitForEtcdUpgrade                      []interface{}
 	PreWaitForControlPlaneToBeUpgraded      func()
 	PreWaitForMachineDeploymentToBeUpgraded func()
+	PreWaitForMachinePoolToBeUpgraded       func()
```

Comment:
Do we need to add this for something in particular? If not I would instead rename `preWaitForMachinePoolToBeUpgraded` to `preWaitForWorkersToBeUpgraded`